src/elements/NavBar/NavBar.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState } from 'react';
+import { useNavigation } from 'react-navigation-hooks';
+import { View, Text, SafeAreaView } from 'react-native';
+import { TouchableOpacity } from 'react-native-gesture-handler';
+import { Button, Icon } from '@elements';
+import { ButtonStyle } from '@elements/Button';
+import * as colors from '@util/colors';
+import Selector from '@elements/NavBar/Selector';
+import HamburgerPopupMenu from '../HamburgerPopupMenu';
+
+import styles from './NavBar.styles';
+
+interface NavBarProps {
+	backDestination?: string;
+	showMenu?: boolean;
+	showBackButton?: boolean;
+	leftButton?: string;
```

Comment:
Set type to `'back' | 'qrCode'` in order to better restrict and explain what this property is expecting?

Reply:
Better yet, maybe use `Pick<IconName, 'back' | 'qrCode>` so that if the icon names change and this isn't updated, typescript will throw an error.

Reply:
I was thinking the same but seems the script was reverted accidentally when I was resolving compatible problem lol. Would update this, thanks for pointing out!

Reply:
Updated by setting type to 'back'|'qrCode', it's a good idea to use Pick but I am not very sure about its mechanism, so let's just keep it simple now and might try this later if the logic there gets more complicated!